client/src/pages/ProfileDetails/ProfileDetails.tsx

```diff
@@ -48,6 +68,22 @@ export default function ProfileDetails(): JSX.Element {
     return () => window.removeEventListener('resize', updateWindowWidth);
   });
 
+  const handleClickOpen = () => {
+    setReviewsDialogOpen(true);
+  };
+
+  const handleClose = () => {
+    setReviewsDialogOpen(false);
+  };
+
+  const addReview = (review: Review) => {
+    if (reviews) {
+      setReviews([review, ...reviews]);
```

Comment:
```suggestion
      setReviews((prevReviews) => ([...prevReviews, review]));
```
Whenever we need to update our current state to next state based on value possed by property just now and all this is happening in async fashion it is good idea to use setState as callback function.